mailbox/plugin/spamassassin/src/main/java/org/apache/james/mailbox/spamassassin/SpamAssassinListener.java

```diff
@@ -69,16 +74,17 @@ public ExecutionMode getExecutionMode() {
 
     @Override
     public void event(Event event) {
+        String username = event.getUser().asString();
         if (event instanceof MessageMoveEvent) {
             MessageMoveEvent messageMoveEvent = (MessageMoveEvent) event;
             if (isMessageMovedToSpamMailbox(messageMoveEvent)) {
                 LOGGER.debug("Spam event detected");
                 ImmutableList<InputStream> messages = retrieveMessages(messageMoveEvent);
-                spamAssassin.learnSpam(messages, messageMoveEvent.getSession().getUser().getUserName());
+                spamAssassin.learnSpam(messages, username);
```

Comment:
I'd rather make learn{Sp,H}am take a user

Reply:
updated in e9bdaf9